packages/@glimmer/core/test/interactive/helper-test.ts

```diff
@@ -43,12 +183,244 @@ module('Modifier Tests', () => {
 
     assert.strictEqual(html, '<h1>helper Hello Rob</h1>', 'the template was rendered');
     assert.equal(count, 1, 'helper did not rerender after unrelated change');
+  });
+
+  test('custom helpers lifecycle (basic)', async assert => {
+    let setupCount = 0;
+    let updateCount = 0;
+    let teardownCount = 0;
+
+    const myHelper = helper(
+      class {
+        args: {
+          positional: [string];
+          named: {
+            greeting: string;
+          };
+        };
+
+        setup(): void {
+          // entangle a value that changes
+          this.args.positional[0];
+          setupCount++;
```

Comment:
Instead of incrementing, can you push into an array (that we assert with deepEqual)?